hilti/src/compiler/cxx/elements.cc

```diff
@@ -436,9 +436,18 @@ std::string cxx::type::Struct::inlineCode() const {
 
     std::string inline_code;
 
-    // Create default constructor.
-    inline_code += fmt("inline %s::%s() {\n%s%s%s}\n\n", type_name, type_name, init_parameters(), init_locals_user(),
-                       init_locals_non_user());
+    // Create default constructor. This initializes user-controlled members
+    // only if there are no struct parameters. If there are, we wouldn't have
+    // access to them here yet and hence some init expressions might not
+    // evaluate. However, in that case we will run through the
+    // parameter-based constructors normally anyways, so don't need this
+    // here.
+    if ( args.size() != 0 )
```

Comment:
```suggestion
    if ( ! args.empty() )
```

Reply:
adding another note to the comment, it's what the if/else does